contrib/golang/filters/http/test/golang_integration_test.cc

```diff
@@ -18,10 +20,66 @@ absl::string_view getHeader(const Http::HeaderMap& headers, absl::string_view ke
   return values[0]->value().getStringView();
 }
 
+class RetrieveDynamicMetadataFilter : public Http::StreamEncoderFilter {
+public:
+  // Http::StreamEncoderFilter
+  Http::Filter1xxHeadersStatus encode1xxHeaders(Http::ResponseHeaderMap&) override {
+    return Http::Filter1xxHeadersStatus::Continue;
+  }
+
+  Http::FilterHeadersStatus encodeHeaders(Http::ResponseHeaderMap&, bool) override {
+    return Http::FilterHeadersStatus::Continue;
+  }
+
+  Http::FilterDataStatus encodeData(Buffer::Instance&, bool) override {
+    EXPECT_EQ("foo", "bar");
```

Comment:
this seems not right, but the tests are pass, a bit weired to me. maybe, this case is not really be tested?

Reply:
turns out I changed it to a response filter but I never waited for the response so the code never got invoked.